internal/storagenode/storagenode_test.go

```diff
@@ -581,3 +590,476 @@ func TestStorageNode_Report(t *testing.T) {
+				require.Equal(t, lastCommittedGLSN, localHWM)
+
+				put(t, dst, 1)
+				status, localHWM = TestSealLogStreamReplica(t, cid, dst.snid, tpid, lsid, lastCommittedGLSN, dst.advertise)
+				require.Equal(t, varlogpb.LogStreamStatusSealing, status)
+				require.Equal(t, lastGLSN(1), localHWM)
+
+				require.Eventually(t, func() bool {
+					syncStatus := TestSync(t, cid, src.snid, tpid, lsid, 0 /*unused*/, src.advertise, varlogpb.StorageNode{
+						StorageNodeID: dst.snid,
+						Address:       dst.advertise,
+					})
+					t.Logf("syncStatus: %s", syncStatus.String())
+					return syncStatus.State == snpb.SyncStateComplete
+				}, 10*time.Second, 100*time.Millisecond)
+
+				status, localHWM = TestSealLogStreamReplica(t, cid, dst.snid, tpid, lsid, lastCommittedGLSN, dst.advertise)
+				require.Equal(t, varlogpb.LogStreamStatusSealed, status)
+				require.Equal(t, lastCommittedGLSN, localHWM)
+
+				snmd, err := dst.getMetadata(context.Background())
+				require.NoError(t, err)
+				require.Equal(t, varlogpb.LogSequenceNumber{
+					LLSN: 3, GLSN: 3,
+				}, snmd.LogStreamReplicas[0].LocalLowWatermark)
+				require.Equal(t, varlogpb.LogSequenceNumber{
+					LLSN: 4, GLSN: 4,
+				}, snmd.LogStreamReplicas[0].LocalHighWatermark)
+			},
+		},
+		{
+			// ver: +-1-+ +-2-+
+			// src:          4
+			// dst:  1 2
+			name: "TrimAndCopy",
+			testf: func(t *testing.T, src, dst *StorageNode) {
+
+				const ver = types.Version(2)
+				lastCommittedGLSN := lastGLSN(ver)
+
+				put(t, src, ver)
+				trim(t, src, 3)
+				status, localHWM := TestSealLogStreamReplica(t, cid, src.snid, tpid, lsid, lastCommittedGLSN, src.advertise)
+				require.Equal(t, varlogpb.LogStreamStatusSealed, status)
+				require.Equal(t, lastCommittedGLSN, localHWM)
+
+				put(t, dst, 1)
+				status, localHWM = TestSealLogStreamReplica(t, cid, dst.snid, tpid, lsid, lastCommittedGLSN, dst.advertise)
+				require.Equal(t, varlogpb.LogStreamStatusSealing, status)
+				require.Equal(t, lastGLSN(1), localHWM)
+
+				require.Eventually(t, func() bool {
+					syncStatus := TestSync(t, cid, src.snid, tpid, lsid, 0 /*unused*/, src.advertise, varlogpb.StorageNode{
+						StorageNodeID: dst.snid,
+						Address:       dst.advertise,
+					})
+					t.Logf("syncStatus: %s", syncStatus.String())
+					return syncStatus.State == snpb.SyncStateComplete
+				}, 10*time.Second, 100*time.Millisecond)
+
+				status, localHWM = TestSealLogStreamReplica(t, cid, dst.snid, tpid, lsid, lastCommittedGLSN, dst.advertise)
+				require.Equal(t, varlogpb.LogStreamStatusSealed, status)
+				require.Equal(t, lastCommittedGLSN, localHWM)
+
+				snmd, err := dst.getMetadata(context.Background())
+				require.NoError(t, err)
+				require.Equal(t, varlogpb.LogSequenceNumber{
+					LLSN: 4, GLSN: 4,
+				}, snmd.LogStreamReplicas[0].LocalLowWatermark)
+				require.Equal(t, varlogpb.LogSequenceNumber{
+					LLSN: 4, GLSN: 4,
+				}, snmd.LogStreamReplicas[0].LocalHighWatermark)
+			},
+		},
```

Comment:
How about adding this case?

ver: +-1-+ +-2-+ +-3-+
src:  ............3 4......5 6
dst:  ...1 2....3 4

Reply:
Good, this is the case you pointed out.
According to the current implementation, the destination has log entries from 1 to 6 after synchronization. If we consider partial trimming, the destination will have log entries from 3 to 6.

Reply:
Added this test case.